services/sharepoint-connector/deploy/helm-charts/sharepoint-connector/templates/config.yaml

```diff
@@ -7,14 +7,27 @@ metadata:
   labels:
     {{- include "chart.labels" . | nindent 4 }}
 data:
-  # Sharepoint config
-  SHAREPOINT_GRAPH_USE_OIDC_AUTH: {{ .Values.connectorConfig.sharepoint.graph.useOidcAuth | quote }}
-  {{- if .Values.connectorConfig.sharepoint.graph.clientId | quote }}
-  SHAREPOINT_GRAPH_CLIENT_ID: {{ .Values.connectorConfig.sharepoint.graph.clientId | quote }}
-  {{- end }}
-  SHAREPOINT_GRAPH_API_URL: {{ .Values.connectorConfig.sharepoint.graph.apiUrl | quote }}
-  SHAREPOINT_GRAPH_TENANT_ID: {{ .Values.connectorConfig.sharepoint.graph.tenantId | quote }}
-  {{- if .Values.connectorConfig.sharepoint.graph.apiRateLimitPerMinute | quote }}
+  # Sharepoint authentication config
+  SHAREPOINT_AUTH_MODE: {{ .Values.connectorConfig.sharepoint.auth.mode | quote }}
+  SHAREPOINT_AUTH_TENANT_ID: {{ .Values.connectorConfig.sharepoint.auth.tenantId | quote }}
+  {{- if eq .Values.connectorConfig.sharepoint.auth.mode "client-secret" }}
```

Comment:
should we fail zod validation if AUTH_MODE is not lowercase when the app starts?